Fix/enhance NMEA & Ublox message polling for GPS

KarateBrot opened the pull request:
This PR fixes some bugs/problems on both NMEA and U-blox protocols to make DOP values work properly. In addition I refactored the code to save some flash and added the U-blox accuracy estimate for position to the GPS solution for use in BF.

The commit names describe in detail what I changed.

Thanks a lot to @ctzsnooze for laying the groundwork for this PR and identifying the problems.

Also thanks a lot to the testers: @benjamind2 and **Sek101**

benjamind2 commented:
Tested, works, provided log GPS_DOP log on discord, from commit `e521d748b2060b7fd0db9ee204a4fb69fb990d2c`.

Used "Flywoo GM8 Mini V2.0 GPS Module" at 115200 baud on a F7X2 SkyStars FC.

Observed very fast lock, but that could be coincidental. Coordinate map, speed, satellites, altitude, distance from home all look good, meet or exceed previous behavior.

KarateBrot commented:
@benjamind2 Thanks for testing! You tested this with the UBlox protocol, so now we need to make sure this also works on the NMEA protocol 👍

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

KarateBrot commented:
> But there is a bug where velocity for GPS Rescue gets now updated by several GPS messages with different values. There is a fix ready for testing. Tomorrow we will know more.

Sek101 confirmed with logs that it is now fixed
